appium/webdriver/extensions/flutter/flutter_commands.py

```diff
@@ -0,0 +1,166 @@
+            opts = {'element': locator, 'timeout': time_out}
+        else:
+            opts = {'locator': locator.to_dict(), 'timeout': time_out}
+
+        self.execute_flutter_command('waitForVisible', opts)
+
+            
+    def wait_for_invisible(self, locator: Union[WebElement, FlutterFinder], time_out: Optional[int] = None) -> None:
+        """
+        Waits for a element to become invisible.
+
+        Args:
+            locator: The element to wait for; can be a WebElement or a FlutterFinder.
+            time_out: Maximum wait time in seconds. Defaults to a predefined timeout if not specified.
+            
+        Returns:
+            None: 
+        """
+        if isinstance(locator, WebElement):
+            opts = {'element': locator, 'timeout': time_out}
+        else:
+            opts = {'locator': locator.to_dict(), 'timeout': time_out}
+        
+        self.execute_flutter_command('waitForAbsent', opts)
+
+    # flutter action commands
+    
+    def perform_double_click(self, element: WebElement, offset: Optional[Tuple[int, int]] = None) -> None:
+        """
+        Performs a double-click on the given element, with an optional offset.
+
+        Args:
+            element: The element to double-click on. This parameter is required.
+            offset: The x and y offsets from the element to click at. If not specified, the click is performed at the element's center.
+            
+        Returns:
+            None:
+        """
+        opts = {'origin': element}
+        if offset:
+            opts['offset'] = {'x': offset[0], 'y': offset[1]}
+        self.execute_flutter_command('doubleClick', opts)
+    
+    def perform_long_press(self, element: WebElement, offset: Optional[Tuple[int, int]] = None) -> None:
+        """
+        Performs a long press on the given element, with an optional offset.
+
+        Args:
+            element: The element to perform the long press on. This parameter is required.
+            offset: The x and y offsets from the element to perform the long press at. If not specified, the long press is performed at the element's center.
+            
+        Returns:
+            None:
+        """
+        opts = {'origin': element}
+        if offset:
+            opts['offset'] = {'x': offset[0], 'y': offset[1]}
+        self.execute_flutter_command('longPress', opts)
+            
+    def perform_drag_and_drop(self, source: WebElement, target: WebElement) -> None:
+        """
+        Performs a drag-and-drop operation from a source element to a target element.
+
+        Args:
+            source: The element to drag from.
+            target: The element to drop onto.
+            
+        Returns:
+            None:
+        """
+        self.execute_flutter_command('dragAndDrop', {'source': source, 'target': target})
+        
+    def scroll_till_visible(self, scroll_to: FlutterFinder, scroll_direction: Optional[str] = 'down', **opts: Any) -> WebElement:
```

Comment:
Prefer to use explicit enum there. See https://docs.python.org/3/library/enum.html

Reply:
Thanks! Added